feat: forward ConfirmDialog class name to overlay

web-padawan opened the pull request:
## Description

Added logic to copy `className` from `vaadin-confirm-dialog` to its overlay element for styling.

Fixes #2909

## Type of change

- Feature

sissbruecker commented:
Instead of introducing the connector, I wonder if we shouldn't rather attempt to solve this in the WC. An alternative might be:
- introduce something like `overlayClass`, `overlayStyle` properties to the dialog components
- have the WCs delegate the property values to `overlay.class`, `overlay.style`
- in the Flow component override the `HasStyle` implementation to use the delegate properties

That would bring the additional benefit that developers using the plain WC, or Hilla, would also have a simple API to modify classes / styles for the overlay.

krissvaa commented:
If class is added when creating dialog and then the dialog is opened without attaching, the class is missing from overlay.
<img width="706" alt="Screenshot 2022-03-22 at 17 44 18" src="https://user-images.githubusercontent.com/101052651/159522436-a17fa725-28c0-44ba-bac4-627601f9094a.png">

https://user-images.githubusercontent.com/101052651/159523293-c1289320-5194-400b-8d44-d0655a39b6de.mp4

web-padawan commented:
> If class is added when creating dialog and then the dialog is opened without attaching, the class is missing from overlay.

Good catch, fixed. Also updated the test intended to cover this to make it actually work as expected.

krissvaa commented on vaadin-confirm-dialog-flow-parent/vaadin-confirm-dialog-flow/src/main/java/com/vaadin/flow/component/confirmdialog/ConfirmDialog.java:
```diff
@@ -126,6 +128,17 @@ public void updateHeight() {
+    }
+
     private boolean autoAddedToTheUi;
 
```
As `getStyle()` is not supported from `HasStyle` with this implementation then probably it should be documented

```suggestion
/**
     * @throws UnsupportedOperationException
     *             Adding styles to dialog overlay are not supported
     */
    @Override
    public Style getStyle() {
        throw new UnsupportedOperationException();
    }
```

web-padawan replied:
Done.
